eng/scripts/InstallationCheck.ps1

```diff
@@ -9,24 +9,33 @@ param (
     [string] $PipelineWorkspace
 )
 
-Write-Host "dotnet new console"
-dotnet new console
+mkdir InstallationCheck
+cd "InstallationCheck"
+
+Write-Host "dotnet new console --no-restore"
+dotnet new console --no-restore
 $localFeed = "$PipelineWorkspace/$ArtifactsDirectory-signed/$Artifact"
-Write-Host "dotnet nuget add source $localFeed"
-dotnet nuget add source $localFeed
 
-$version = (Get-ChildItem "$localFeed/*.nupkg" -Exclude "*.symbols.nupkg" -Name).replace(".nupkg","").replace("$Artifact.","")
+$version = (Get-Content "$PipelineWorkspace/$ArtifactsDirectory-signed/PackageInfo/$Artifact.json" | ConvertFrom-Json).Version
+
 Write-Host "dotnet add package $Artifact --version $version --no-restore"
 dotnet add package $Artifact --version $version --no-restore
 if ($LASTEXITCODE) {
-    exit $LASTEXITCODE
+  exit $LASTEXITCODE
 }
 
-Write-Host "dotnet nuget locals all --clear"
-dotnet nuget locals all --clear
-
-Write-Host "dotnet restore -s https://api.nuget.org/v3/index.json -s $localFeed --no-cache --verbosity detailed"
-dotnet restore -s https://api.nuget.org/v3/index.json -s $localFeed --no-cache --verbosity detailed
-if ($LASTEXITCODE) {
-    exit $LASTEXITCODE
-}
\ No newline at end of file
+while ($retries++ -lt 30) {
+  Write-Host "dotnet restore -s https://api.nuget.org/v3/index.json -s $localFeed --no-cache --verbosity detailed"
+  dotnet restore -s https://api.nuget.org/v3/index.json -s $localFeed --no-cache --verbosity detailed
```

Comment:
Looks like you decided to not use `--packages` option. Thinking about it a little more since we are setting the exact version to the unreleased version I think that is probably ok.